src/vm/wren_compiler.c

```diff
@@ -3416,20 +3416,27 @@ void definition(Compiler* compiler)
 ObjFn* wrenCompile(WrenVM* vm, ObjModule* module, const char* source,
                    bool isExpression, bool printErrors)
 {
+  // Skip potential UTF-8 BOM
+  size_t sourceOffset = 0;
+  if (source[0] == (char) 0xef && source[1] == (char) 0xbb && source[2] == (char) 0xbf)
```

Comment:
This can be crunched down even further to `if (strncmp(source, "\xEF\xBB\xBF", 3) == 0) source += 3;` with the added bonus of not needing to do all the offsets below.

Reply:
@jclc that's not a const pointers, that's a pointer to a const. A const pointer would be `char *const source`.

Reply:
Bonus aside you can also have turbo-`const` pointers taking the form `const <type>* const` that have both properties.